src/core/agent/agent.types.ts

```diff
@@ -157,8 +157,9 @@ interface KeriaNotification {
   id: string;
   createdAt: string;
   a: Record<string, unknown>;
-  multisigId?: string | undefined;
+  multisigId?: string;
   connectionId: string;
+  linkedGroupRequests?: Record<string, boolean>;
```

Comment:
Since we default to `{}` - can we remove the `?` here?

Reply:
I think in case single-sig, linkedGroupRequests will not be used. So is setting the default value for notification to {} redundant?

Reply:
```
  linkedGroupRequests?: Record<string, boolean>;
  static readonly type = "NotificationRecord";
  readonly type = NotificationRecord.type;

  constructor(props: NotificationRecordStorageProps) {
    super();
    if (props) {
      this.id = props.id ?? uuidv4();
      this.createdAt = props.createdAt ?? new Date();
      this.a = props.a;
      this.route = props.route;
      this.read = props.read;
      this.multisigId = props.multisigId ?? undefined;
      this.connectionId = props.connectionId;
      this._tags = props.tags ?? {};
      this.linkedGroupRequests = props.linkedGroupRequests ?? {};
    }
  }
```

Here we are already always setting it to `{}` if it's undefined. So we can remove the `?` from the first line above, and from `agent.types.ts` as I suggested in this comment.

It's not used for single sig but it doesn't matter - we already have `multisigId` to differentiate between single and multi-sig. If we keep it set as `{}` at minimum always, it will make the rest of our code simpler (no more undefined checks etc)